UnitTests/CommonTestUtils/ConfigureJoinableTaskFactoryAttribute.cs

```diff
@@ -56,10 +70,9 @@ public void AfterTest(ITest test)
             {
                 try
                 {
-                    using (var cts = new CancellationTokenSource(AsyncTestHelper.UnexpectedTimeout))
-                    {
-                        ThreadHelper.JoinableTaskContext?.Factory.Run(() => ThreadHelper.JoinPendingOperationsAsync(cts.Token));
-                    }
+                    // Wait for eventual pending operations triggered by the test.
+                    // On timeout, only log this but do not let fail the test.
+                    AsyncTestHelper.WaitForPendingOperations(AsyncTestHelper.UnexpectedTimeout, throwOnTimeout: false);
```

Comment:
❗️ Failing the test on timeout is absolutely essential. The timeout exists _only_ to avoid test hangs. It is not used to make an incorrectly behaved test pass.

Reply:
Then we must wrap `OperationCanceledException` at least, because it cancels all tests that folllow.

How can we find out why the pending operations of `ScriptRunnerTests` do not finish on AppVeyor occasionally? `Tasks` cannot be analyzed if the debugger is attached afterwards.

Reply:
```patch
-_threadException?.Throw();
+// reset _threadException to null, and throw if it was set during the current test
+Interlocked.Exchange(ref _threadException, null)?.Throw();
```

Reply:
> Tasks cannot be analyzed if the debugger is attached afterwards.

I'd start with this: https://github.com/microsoft/vs-threading/blob/master/doc/async_hang.md

We should also be able to use [`JoinableTaskContextNode`](https://github.com/microsoft/vs-threading/blob/b680de515d6210c4028598b988187c3aed33aabc/src/Microsoft.VisualStudio.Threading/JoinableTaskContextNode.cs) to override the `OnHangDetected` method to obtain further information about delays.

Reply:
`OnHangDetected` works but doesn't give much information. `HangDetails` is hidden by `JoinableTaskContextNode`.
`Debug.Break` has no effect on AppVeyor.
How can I force the creation of a crash dump in order to analyze it offline?

Reply:
WinDbg does not display very much:
```
(86c.768): Break instruction exception - code 80000003 (first chance)
ntdll!DbgBreakPoint:
00007ffd`e18d3150 cc              int     3
0:019> .load c:\Users\appveyor\Downloads\SosThreadingTools_x64.dll
0:019> !dumpasync
e0434352 Exception in c:\Users\appveyor\Downloads\SosThreadingTools_x64.dll.dumpasync debugger extension.
      PC: 00007ffd`de779159  VA: 00000000`00000000  R/W: 80070002  Parameter: 00000000`00000000
```